cmds/rm/rm_test.go

```diff
@@ -38,58 +39,158 @@ var tests = []makeit{
 	},
 }
 
-// initial files and folders (for testing)
+/* Tests2 contains files, directories, links, and block devices */
+var tests2 = []makeit{
+	{
+		n: "file.txt",
+		m: 0000,
+		s: "",
+	},
+	{
+		n: "dir",
+	},
+	{
+		n: "dirWfiles",
+	},
+	{
+		n: "block",
+	},
+	{
+		n: "symFile",
+	},
+}
+
 func setup() (string, error) {
 	fmt.Println(":: Creating simulating data...")
 	d, err := ioutil.TempDir(os.TempDir(), "hi.dir")
 	if err != nil {
 		return "", err
 	}
 
-	tmpdir := path.Join(d, "hi.sub.dir")
-	if err := os.Mkdir(tmpdir, 0777); err != nil {
-		return "", err
-	}
-
+	/* Creates some files with content */
 	for i := range tests {
 		if err := ioutil.WriteFile(path.Join(d, tests[i].n), []byte("Go is cool!"), tests[i].m); err != nil {
 			return "", err
 		}
 	}
 
+	/* Creates empty file and directory s */
+	if err := ioutil.WriteFile(path.Join(d, tests2[0].n), nil, tests2[0].m); err != nil {
+		return "", err
+	}
+	if ioutil.TempDir(d, tests2[1].n); err != nil {
+		return "", err
+	}
+
+	/* Create file, symlink, and block device in a new directory */
+	newD, err := ioutil.TempDir(d, tests2[2].n)
+	if err != nil {
+		return "", err
+	}
+	if err := ioutil.WriteFile(path.Join(newD, tests2[0].n), []byte(""), tests2[0].m); err != nil {
+		return "", err
+	}
+
+	if err := os.Symlink(path.Join(d, tests2[0].n), path.Join(newD, "symlink")); err != nil {
+		return "", err
+	}
+	/* Does not work on Travis
+	if err := syscall.Mknod(path.Join(newD, tests2[3].n), 0777, 64); err != nil {
+		return "", err
+	}
+	*/
 	return d, nil
 }
 
-// without any flag
+func printFiles(d string) (filenames []string, err error) {
+	var nameArray []string
+	fileNames, err := ioutil.ReadDir(d)
+	if err != nil {
+		return nil, err
+	}
+	for _, file := range fileNames {
+		fmt.Println(file.Name())
+		nameArray = append(nameArray, path.Join(d, file.Name()))
+	}
+	return nameArray, nil
+}
+
+// TEST 1
+// Flags: -v; only delete files from tests
+
 func Test_rm_1(t *testing.T) {
+	fmt.Println("TEST 1:")
 	d, err := setup()
 	if err != nil {
-		t.Fatal("Error on setup of the test: creating files and folders.")
+		t.Fatalf("Error on setup of the test: creating files and folders: %s", err)
 	}
-	defer os.RemoveAll(d)
-
 	fmt.Println("== Deleting files and empty folders (no args) ...")
 	files := []string{path.Join(d, "hi1.txt"), path.Join(d, "hi2.txt"), path.Join(d, "go.txt")}
-
-	flags.v = true
-	if err := rm(files); err != nil {
+	var flags rmFlags
+	flags.verbose = true
+	if err := rm(files, flags); err != nil {
 		t.Error(err)
 	}
+	os.RemoveAll(d)
 }
 
-// using r flag
+// TEST 2
+// Flags: -v -r; only delete files from test1
 func Test_rm_2(t *testing.T) {
+	fmt.Println("TEST 2:")
```

Comment:
You should remove all fmt Prints in the final version. If all tests pass, the only thing we should see printed is golang's PASS. Prints are only useful if we fail, so they can be printed by t.Error and t.Fatal. 

What do you think?